webplugin/js/app/mck-sidebox-1.0.js

```diff
@@ -5485,9 +5428,95 @@ var userOverride = {
                             return false;
                         }
                     }
-                    _this.hideSendButton();
-                    Kommunicate.typingAreaService.showMicIfRequiredWebAPISupported();
-                    _this.sendMessage(messagePxy);
+                    //If the field is a form field then validate the input, update user details before sending the message
+                    if ($mck_text_box.data('fieldType')) {
+                        //If the field has a regex validation then validate the input otherwise skip validation
+                        if($mck_text_box.data('validation')){
+                            var regexForm = $mck_text_box.data(
+                                'validation'
+                            );
+                            regexForm = new RegExp(regexForm);
+                            //If the input does not match the regex validation then show an error message for 2 seconds.
+                            if (!regexForm.test(message)) {
+                                $applozic('#mck-form-field-error-alert').html($mck_text_box.data('errorMessage'));
+                                $applozic('#mck-form-field-error-alert-box')
+                                    .removeClass('n-vis')
+                                    .addClass('vis');
+                                setTimeout(function () {
+                                    $applozic('#mck-form-field-error-alert-box')
+                                        .removeClass('vis')
+                                        .addClass('n-vis');
+                                }, 2000);
+                                return false;
+                            }
+                        }
+                        if($mck_text_box.data('updateUserDetails')){
+                            // If the field is a form field and the user details need to be updated then update the user details
+                            var fieldVal = $mck_text_box.data('field');
+                            var userUpdateField = {};
+                            userUpdateField[fieldVal] = message;
+                            if (
+                                $mck_text_box.data('fieldType') === 'EMAIL' ||
+                                $mck_text_box.data('fieldType') === 'NAME' ||
+                                $mck_text_box.data('fieldType') ===
+                                    'PHONE_NUMBER'
+                            ) {
+                                mckContactService.updateUser({
```

Comment:
Also, see we can avoid this update user call altogether. 
There are 2 issues
1. when bot asks for name and instead of only name, you say "my name is abhishek" then "my name is abhishek" gets updated as name of the user
2. if updateUser() call fails then we again face the same issue.
```suggestion
                                mckContactService.updateUser({
```